torch/lib/c10d/CMakeLists.txt

```
@@ -77,14 +78,17 @@ endif()
 add_library(c10d STATIC ${C10D_SRCS})
 set_property(TARGET c10d PROPERTY POSITION_INDEPENDENT_CODE ON)
 set_property(TARGET c10d PROPERTY CXX_STANDARD 14)
-target_compile_options(c10d PUBLIC
-  -Wall
-  -Wextra
-  -Wno-unused-parameter
-  -Wno-missing-field-initializers
-  -Wno-write-strings
-  -Wno-unknown-pragmas
-  )
+
+if(NOT WIN32)
```

This one is actually compiler specific, rather than platform (i.e. if someone will try to compile this code on Windows but using CLANG compiler we want to pass -Wall ... flags, don't we?
```suggestion
if(NOT MSVC)
```